### PR TITLE
Unicode:UCD Clarify pod for num()

### PR DESCRIPTION
**Pull request by khwilliamson**
Add example of how it handles numbers that aren't decimal positional.

I thought it would clarify things to expand and correct the flawed example pointed out in GH #23003.

* This set of changes does not require a perldelta entry.

### CODE REVIEW

File: lib/Unicode/UCD.pm [hunk 1]
```diff
@@ -2496,6 +2496,32 @@ match them.  A single-character string containing one of these digits will
...
+
+ my $rumi_one = num("\N{RUMI DIGIT ONE}");
+ my $rumi_two = num("\N{RUMI DIGIT TWO}");
+ my $rumi_twenty = num("\N{RUMI NUMBER TWENTY}");
+ say "$rumi_one $rumi_two $rumi_twenty";     # 1 2 20
+
+Because these do not follow modern decimal positional notation, stringing more
+than one of these together doesn't mean what you likely would think it means.
+So, C<num> correctly returns C<undef> if you try.  If you add a second
+parameter to get returned to you the initial legal numeric length, it returns
```
Comment (tonycoz) [inline lib/Unicode/UCD.pm]: > If you add a second parameter to get returned to you the initial legal numeric length  it returns indicating that to get a numeric value  you'd have to stop after just the first character in the string.

This is pretty clumsy.

Maybe:

> If you request the length of the valid initial substring in this case, that length would be one, since only the first character represents a valid numeric value.

though I'm not entirely happy with that either.

**Review by jkeenan — COMMENTED** [review]
Should we add some unit tests to `lib/Unicode/UCD.t` (or some other file) to demonstrate the code added to the POD?  (I couldn't find a `.t` file that contained a match for `m/RUMI/t`.)

**Comment by khwilliamson** [comment]
This is a documentation PR only.  I chose Rumi because I had recently researched it, and the details were fresh in my mind.  There are tests for similar situations already in `UCD.t`.  The ones for New Tai Lue, for example.

**Comment by jkeenan** [comment]
> This is a documentation PR only. I chose Rumi because I had recently researched it, and the details were fresh in my mind. There are tests for similar situations already in `UCD.t`. The ones for New Tai Lue, for example.

Okay; once @tonycoz's concerns have been satisfied, this p.r. is good to go. Thanks.

**Comment by khwilliamson** [comment]
I addressed those concerns
